code/modules/food_and_drinks/restaurant/generic_venues.dm

```diff
@@ -22,8 +22,13 @@
 	var/obj/item/object_to_order = customer_data.get_order(src)
 
 	. = object_to_order
-
-	customer_pawn.say(order_food_line(object_to_order))
+	var/line_to_say = order_food_line(object_to_order)
+	if(prob(1) || SSevents.holidays && SSevents.holidays[APRIL_FOOLS])
```

Comment:
1. This precedence doesn't make any sense at all. If there's no holidays, then it's just a 1% chance to runtime (because prob(1) will pass, then SSevents.holidays[APRIL_FOOLS] will runtime).
2. This should only be for American bots.

Reply:
Keep it to April Fools, also.

Reply:
If the holidays subsystem doesn't exist, we have bigger problems because that means subsystems didn't init.

Reply:
is it
`prob(1) OR (SSevents.holidays AND SSevents.holidays[APRIL_FOOLS])`
or
`(prob(1) OR SSevents.holidays) AND SSevents.holidays[APRIL_FOOLS]`

Reply:
Why are you checking `SSevents.holidays` if you're so sure it's never null?

Reply:
to dodge a runtime

Reply:
if it's null don't we want to know?

Reply:
> If the holidays subsystem doesn't exist, we have bigger problems because that means subsystems didn't init.
----------
> to dodge a runtime if it is null

I can't tell if you're wanting to play it safe or not care about a runtime, regardless though this probability is way too high for all year. I'd do this line like
```
var/probability = 0.1 * ( SSevents.holidays[APRIL_FOOLS] ? 1 : 0.1 )
if(prob(probability))
```

Also this code should be in `order_food_line()` and not just sitting out here in the main `order_food()` proc